language/move-stdlib/src/natives/debug.rs

```diff
@@ -3,20 +3,537 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::natives::helpers::make_module_natives;
+#[cfg(feature = "testing")]
+use move_binary_format::errors::PartialVMError;
 use move_binary_format::errors::PartialVMResult;
+use move_core_types::{account_address::AccountAddress, gas_algebra::InternalGas};
+#[cfg(feature = "testing")]
 use move_core_types::{
-    account_address::AccountAddress, gas_algebra::InternalGas, language_storage::TypeTag,
+    language_storage::{StructTag, TypeTag},
+    value::{MoveStruct, MoveStructLayout, MoveTypeLayout, MoveValue},
+    vm_status::StatusCode,
 };
 use move_vm_runtime::native_functions::{NativeContext, NativeFunction};
-use move_vm_types::values::VectorRef;
-#[allow(unused_imports)]
-use move_vm_types::values::{values_impl::debug::print_reference, Reference, StructRef};
+#[cfg(feature = "testing")]
+use move_vm_types::values::Reference;
 #[allow(unused_imports)]
 use move_vm_types::{
     loaded_data::runtime_types::Type, natives::function::NativeResult, pop_arg, values::Value,
 };
 use smallvec::smallvec;
 use std::{collections::VecDeque, sync::Arc};
+#[cfg(feature = "testing")]
+use std::{
+    fmt,
+    fmt::{Error, Write},
+};
+
+#[cfg(feature = "testing")]
+fn is_string_struct_tag(struct_tag: &StructTag, move_std_addr: &AccountAddress) -> bool {
```

Comment:
I think this might want be useful as a function in `impl StructTag` [here](https://github.com/move-language/move/blob/main/language/move-core/types/src/language_storage.rs), since there is a decent amount of code that wants to special-case strings.

Reply:
Will move!